### PR TITLE
fix(DataSpreadsheet): Add null check to `setActiveCellContent`

### PR DESCRIPTION
**Pull request by AustinGitHub**
Add null check

Closes # https://github.com/carbon-design-system/ibm-products/issues/5270

#### What did you change?

I added null check 

#### How did you test and verify your work?

I verified using comparing how the logic works for when cell is being deleted in `handleCellDeletion.js` it is then set to `null` so I added that same check here.

### CODE REVIEW

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx [hunk 1]
```diff
@@ -291,7 +291,11 @@ export let DataSpreadsheet = React.forwardRef(
                   activeCellCoordinates?.column
                 ]
               : null;
-          setActiveCellContent(activeCellFullData.render('Cell'));
+          if (activeCellFullData) {
+            setActiveCellContent(activeCellFullData.render('Cell'));
+          } else {
+            setActiveCellContent(null);
+          }
```
Comment (AustinGitHub) [inline packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:298]: What is odd is not sure why we allow `activeCellFullData` to be `null` in this condition block here but then try to do a `.render('cell')` on it. Seems a use case was missed here. So my fix should patch that, thanks!

Screenshot 2024-05-20 at 1 39 49 PM

**Comment by AustinGitHub** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by AustinGitHub** [comment]
recheck

**Comment by elycheea** [comment]
@AustinGitHub The commit history shows two different IDs. Usually when we see this happen, it’s because the user of the commit is tied to your internal GitHub account. You can check the email associated with these commits with `git show <commit-hash>`. You can either amend the commits (which can sometimes feel dirtier) or create a new PR with the changes (usually easier if changes are small).

> image
> image

Also no need to add yourself to the contributors file. We usually use the allcontributors bot to add you both to the `.contributorsrc` and the actual contributor table.

**Comment by AustinGitHub** [comment]
thanks, I will try to fix it

**Comment by AustinGitHub** [comment]
I made a new PR https://github.com/carbon-design-system/ibm-products/pull/5305 thanks

**Pull request closed without merging**